src/common/CMakeLists.txt

```diff
@@ -97,7 +97,8 @@ set(common_srcs
   utf8.c
   util.cc
   version.cc
-  xattr.c)
+  xattr.c
+  ${CMAKE_SOURCE_DIR}/src/mds/metrics/client/Types.cc)
```

Comment:
I've not incorporated this yet.

Reply:
@batrick sorry for the latency. i am actually surprised that we are putting mds specific stuff into ceph-common. how is it used by other components? can we include it in `mds_srcs` , and link against `mds` instead ?

Reply:
>  i am actually surprised that we are putting mds specific stuff into ceph-common

That's just where it is pending the result of this discussion :)

> how is it used by other components?

client and MDS

> can we include it in mds_srcs , and link against mds instead ?

Perhaps just move it to a header file in `src/include` and get rid of `Types.cc`? I don't like the idea of the client linking to code in the MDS but we kinda already do this for `src/tools/cephfs`. So it wouldn't be too big a departure from what we already do.

Reply:
agreed. we are not doing great in this area.. probably we can avoid making this worse by going  further in this direction?

even though we have libceph-common.so, libceph-common.a is still statically linked into quite a few tests and ceph tools. i really want to avoid putting more objects into that bucket. for couple reasons, it takes more than 1 hour to upload the debuginfo now. and probably more important is that it just does not look/sound right ...

Reply:
Does moving the entire thing Types.cc/Types.h into a header in /src/include  sound good?

Reply:
Actually, the header could go in `src/mds`, we already link to some MDS headers in the client.

Reply:
> Does moving the entire thing Types.cc/Types.h into a header in /src/include sound good?

sounds good to me.